koku/masu/database/trino_sql/gcp/openshift/populate_daily_summary/1_resource_matching_by_cluster.sql

```diff
@@ -44,7 +44,7 @@ WITH cte_gcp_resource_names AS (
     FROM hive.{{schema | sqlsafe}}.gcp_line_items_daily
     WHERE source = {{cloud_provider_uuid}}
         AND year = {{year}}
-        AND month = {{month}}
+        AND month in ({{month}}, {{previous_month}})
```

Comment:
Do you see any value in dynamically decided when to look into two partitions. Right now we have it hard coded to always look in both partitions:

```
select array_agg(distinct month) from gcp_line_items_daily where DATE(usage_start_time) > DATE('2025-06-28') and DATE(usage_start_time) <=
 DATE('2025-07-02') and year='2025' and source='5e906f11-43d0-4d5c-9fc3-5b777225a536';
```

```
select array_agg(distinct month) from gcp_line_items_daily where DATE(usage_start_time) > DATE('2025-07-01') and DATE(usage_start_time) <=
               ->  DATE('2025-07-02') and year='2025' and source='5e906f11-43d0-4d5c-9fc3-5b777225a536';
```

This could help prevent us from querying both partitions when it is not necessary.

Reply:
Darn I had auto-merge on :face_exhaling: Here is a PR for that though! Does make sense to do this for sure https://github.com/project-koku/koku/pull/5669